metabase/metabase
# Split repro for #13641 in two versions: "remapped" and "default"

### Status
MERGED

### What does this PR accomplish?
- Splits repro for #13641 into 2 versions
    - "Remapped" (display values remapped to the FK) - which is now passing (changes introduced in #14695)
    - "Default" version without remapping - still failing at cc348c8c9f3e6bcd0f3780bdbed814fafd5c47cb

See @flamber's [comment](https://github.com/metabase/metabase/pull/14695#pullrequestreview-584885256) for an additional context.

### How to test this manually?
- `yarn test-cypress-open --testFiles enterprise/frontend/test/metabase-enterprise`
- `enterprise/frontend/test/metabase-enterprise/sandboxes/sandboxes.cy.spec.js`
- You can isolate this test by adding.only() to the test
```javascript
it.only(`${test.toUpperCase()} version:\n drill-through should work on implicit joined tables with sandboxes (metabase#13641)`, () => {...})
```
- Then comment out or temporarily unskip the `cy.runnable().skip()` part to let the test run for both scenarios
```javascript
test === "remapped"
  ? (cy.log("**-- Remap Product ID's display value to `title` --**"),
    remapDisplayValueToFK(ORDERS.PRODUCT_ID, PRODUCTS.TITLE))
  : cy.state("runnable").skip();
```
- The "default" version of the test should fail until the related issue is fixed (please see the screenshot below)

### Additional notes:
- Once the issue is fixed, please enable both test cases
- Make sure both are passing and
- Merge them together with the fix

### Screenshots:
1. Local testing only with both tests enabled
![image](https://user-images.githubusercontent.com/31325167/107288404-b2436580-6a63-11eb-83b6-a878b622f45d.png)

2. "Default" version of the test skipped
![image](https://user-images.githubusercontent.com/31325167/107288511-d99a3280-6a63-11eb-834a-623e2cea49ea.png)
